server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -4569,6 +4847,59 @@ protected boolean savePublicIPRange(final String startIP, final String endIP, fi
         return problemIps != null && problemIps.size() == 0;
     }
 
+    @DB
+    protected boolean updatePublicIPRange(final String newStartIP, final String currentStartIP, final String newEndIP, final String currentEndIP, final long zoneId, final long vlanDbId, final long sourceNetworkid, final long physicalNetworkId, final boolean isRangeForSystemVM, final Boolean forSystemVms) {
```

Comment:
can you here make a javadoc with what a `true` or `false` return value mean and what the side effects are in terms of DB contents?